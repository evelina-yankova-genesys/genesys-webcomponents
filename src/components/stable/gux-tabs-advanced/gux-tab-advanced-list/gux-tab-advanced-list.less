@import (reference) '../../../../style/color.less';
@import (reference) '../../../../style/mixins.less';
@import (reference) '../../../../style/spacing.less';
@import (reference) '../../../../style/typography.less';

// Variables part
@gux-tabs-background: @gux-grey-60;
@gux-off-tab-fill: @gux-grey-40;

gux-tab-advanced-list {
  .gux-tab-container {
    display: flex;
    align-items: flex-end;
    width: 100%;
    height: 50px;
    margin-bottom: @spacing-medium;
    overflow-x: auto;
    overflow-y: hidden;
    background-color: @gux-grey-90;
    border-bottom: 1px solid @gux-grey-50;
    border-radius: 4px 4px 0 0;

    .action-button-container {
      display: flex;
      border-radius: 4px;

      .add-tab-button {
        min-width: 35px;
        height: 50px;
        padding: 0;
        margin-left: @spacing-3xs;
        color: @gux-black-50;
        background-color: @gux-grey-60;
        border: none;
        border-radius: 4px;
        transition: color 0.25s;

        [slot='add-tab'] {
          display: inline-flex;
          align-items: center;
          padding: 0 @spacing-small;
          white-space: nowrap;
          .body-font-bold();

          gux-icon {
            padding-right: @spacing-xs;
          }
        }

        &:disabled {
          opacity: 0.5;
        }

        &:hover {
          background-color: @gux-grey-50;
        }

        &:active {
          background-color: @gux-grey-40;
        }

        &:focus {
          outline: none;
          box-shadow: inset 0 0 0 3px @gux-blue-90;
        }

        &:focus-visible {
          border-radius: 5px;
          outline: none;
          box-shadow: inset 0 0 0 3px @gux-blue-90;
        }

        gux-icon {
          width: 16px;
          height: 16px;
        }
      }

      .gux-scroll-button-container {
        display: flex;
        border-radius: 4px;

        .gux-scroll-button {
          display: flex;
          align-items: center;
          justify-content: center;
          width: 28px;
          height: 50px;
          padding: 0;
          color: @gux-black-50;
          background-color: @gux-grey-60;
          border: none;
          border-radius: 4px;
          transition: color 0.25s;

          &:focus-visible {
            border: 3px solid @gux-blue-90;
            border-radius: 5px;
            outline: none;
          }

          gux-icon {
            width: 16px;
            height: 16px;
          }

          &:focus {
            border: 3px solid @gux-blue-90;
            outline: none;
          }

          &:hover {
            background-color: @gux-grey-50;
          }

          &:active {
            background-color: @gux-grey-40;
          }

          &:focus-visible {
            .focus-ring();
          }
        }
      }
    }

    .gux-scrollable-section {
      display: flex;
      align-items: flex-end;
      height: 50px;
      overflow-x: auto;
      scroll-behavior: smooth;
      scrollbar-width: none;

      &::-webkit-scrollbar {
        height: 0;
      }
    }
  }

  .gux-sr-only {
    .gux-sr-only();
  }
}
