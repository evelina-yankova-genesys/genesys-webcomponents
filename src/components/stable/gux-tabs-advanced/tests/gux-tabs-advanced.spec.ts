import { newSpecPage } from '@stencil/core/testing';
import MutationObserver from 'mutation-observer';

import { GuxTabsAdvanced } from '../gux-tabs-advanced';
import { GuxTabAdvanced } from '../gux-tab-advanced/gux-tab-advanced';
import { GuxTabAdvancedList } from '../gux-tab-advanced-list/gux-tab-advanced-list';
import { GuxTabAdvancedPanel } from '../gux-tab-advanced-panel/gux-tab-advanced-panel';

const components = [
  GuxTabsAdvanced,
  GuxTabAdvanced,
  GuxTabAdvancedList,
  GuxTabAdvancedPanel
];
const html = `
  <gux-tabs-advanced lang="en" id="interactive">
    <gux-tab-advanced-list slot="tab-list" show-new-tab-button="true">
      <gux-tab-advanced tab-id="1-1">
        <gux-icon icon-name="lock" decorative="true"></gux-icon>
        Hello World
        <gux-list slot="dropdown-options">
          <gux-list-item>
            <gux-icon icon-name="edit" decorative="true"></gux-icon>
            Edit
          </gux-list-item>
          <gux-list-item>
            <gux-icon icon-name="clone" decorative="true"></gux-icon>
            Clone
          </gux-list-item>
          <gux-list-item>
            <gux-icon icon-name="share" decorative="true"></gux-icon>
            Share
          </gux-list-item>
          <gux-list-item>
            <gux-icon icon-name="download" decorative="true"></gux-icon>
            Download
          </gux-list-item>
        </gux-list>
      </gux-tab-advanced>
      <gux-tab-advanced tab-id="1-2">
        <gux-icon icon-name="user-directory" decorative="true"></gux-icon>
        Hello World 2 long long long long long
      </gux-tab-advanced>
      <gux-tab-advanced gux-disabled tab-id="1-3">
      Hello World 3
    </gux-tab-advanced>
    <div slot="add-tab">
      <gux-icon icon-name="file-preview" decorative></gux-icon>
      Add View
    </div>
    </gux-tab-advanced-list>
    <gux-tab-advanced-panel tab-id="1-1">
      <span>Tab content 1</span>
    </gux-tab-advanced-panel>
    <gux-tab-advanced-panel tab-id="1-2">
      <span>Tab content 2</span>
    </gux-tab-advanced-panel>
    <gux-tab-advanced-panel tab-id="1-3">
    <span>Tab content 3</span>
    </gux-tab-advanced-panel>
  </gux-tabs-advanced>
`;

const language = 'en';

describe('gux-tabs-advanced', () => {
  beforeEach(async () => {
    (
      global as NodeJS.Global & {
        MutationObserver: any;
      }
    ).MutationObserver = MutationObserver;
  });
  describe('#render', () => {
    it(`should render as expected`, async () => {
      const page = await newSpecPage({
        components,
        html,
        language
      });
      expect(page.rootInstance).toBeInstanceOf(GuxTabsAdvanced);
      expect(page.root).toMatchSnapshot();
    });
  });
});
