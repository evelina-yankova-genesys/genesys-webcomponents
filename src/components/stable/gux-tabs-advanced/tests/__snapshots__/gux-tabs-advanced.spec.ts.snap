// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-tabs-advanced #render should render as expected 1`] = `
<gux-tabs-advanced id="interactive" lang="en">
  <mock:shadow-root>
    <div class="gux-tabs">
      <slot name="tab-list"></slot>
      <div>
        <slot></slot>
      </div>
    </div>
  </mock:shadow-root>
  <gux-tab-advanced-list show-new-tab-button="true" slot="tab-list">
    <!---->
    <span aria-live="polite" class="gux-aria-live-region gux-sr-only"></span>
    <div class="gux-tab-container">
      <div class="action-button-container">
        <div class="gux-scroll-button-container"></div>
      </div>
      <div aria-owns="gux-1-1-tab gux-1-2-tab gux-1-3-tab" class="gux-scrollable-section" role="tablist">
        <gux-tab-advanced tab-id="1-1">
          <!---->
          <div class="gux-dropdown-options gux-show gux-tab">
            <button aria-controls="gux-1-1-panel" aria-disabled="false" aria-selected="false" class="gux-tab-button" id="gux-1-1-tab" role="tab" tabindex="-1" type="button">
              <gux-tooltip-title>
                <span>
                  <gux-icon decorative="true" icon-name="lock"></gux-icon>
                  Hello World
                </span>
              </gux-tooltip-title>
            </button>
            <button aria-expanded="false" class="gux-tab-options-button" id="gux-i" tabindex="-1" type="button">
              <gux-icon icon-name="menu-kebab-vertical" screenreader-text="Tab Options for Hello World"></gux-icon>
            </button>
            <gux-popover-list closeonclickoutside="" for="gux-i" hidden="" position="top-end">
              <div class="gux-dropdown-option-container">
                <gux-list slot="dropdown-options">
                  <gux-list-item>
                    <gux-icon decorative="true" icon-name="edit"></gux-icon>
                    Edit
                  </gux-list-item>
                  <gux-list-item>
                    <gux-icon decorative="true" icon-name="clone"></gux-icon>
                    Clone
                  </gux-list-item>
                  <gux-list-item>
                    <gux-icon decorative="true" icon-name="share"></gux-icon>
                    Share
                  </gux-list-item>
                  <gux-list-item>
                    <gux-icon decorative="true" icon-name="download"></gux-icon>
                    Download
                  </gux-list-item>
                </gux-list>
              </div>
            </gux-popover-list>
          </div>
        </gux-tab-advanced>
        <gux-tab-advanced tab-id="1-2">
          <!---->
          <div class="gux-show gux-tab">
            <button aria-controls="gux-1-2-panel" aria-disabled="false" aria-selected="false" class="gux-tab-button" id="gux-1-2-tab" role="tab" tabindex="-1" type="button">
              <gux-tooltip-title>
                <span>
                  <gux-icon decorative="true" icon-name="user-directory"></gux-icon>
                  Hello World 2 long long long long long
                </span>
              </gux-tooltip-title>
            </button>
          </div>
        </gux-tab-advanced>
        <gux-tab-advanced gux-disabled="" tab-id="1-3">
          <!---->
          <div class="gux-disabled gux-show gux-tab">
            <button aria-controls="gux-1-3-panel" aria-disabled="true" aria-selected="false" class="gux-tab-button" id="gux-1-3-tab" role="tab" tabindex="-1" type="button">
              <gux-tooltip-title>
                <span>
                  Hello World 3
                </span>
              </gux-tooltip-title>
            </button>
          </div>
        </gux-tab-advanced>
      </div>
      <div class="action-button-container">
        <div class="gux-scroll-button-container"></div>
        <button class="add-tab-button" title="Add View">
          <div slot="add-tab">
            <gux-icon decorative="" icon-name="file-preview"></gux-icon>
            Add View
          </div>
          <slot-fb hidden="" name="add-tab">
            <gux-icon decorative="" icon-name="add"></gux-icon>
          </slot-fb>
        </button>
      </div>
    </div>
  </gux-tab-advanced-list>
  <gux-tab-advanced-panel tab-id="1-1">
    <!---->
    <div aria-labelledby="gux-1-1-tab" aria-live="assertive" hidden="" id="gux-1-1-panel" role="tabpanel" tabindex="0">
      <span>
        Tab content 1
      </span>
    </div>
  </gux-tab-advanced-panel>
  <gux-tab-advanced-panel tab-id="1-2">
    <!---->
    <div aria-labelledby="gux-1-2-tab" aria-live="assertive" hidden="" id="gux-1-2-panel" role="tabpanel" tabindex="0">
      <span>
        Tab content 2
      </span>
    </div>
  </gux-tab-advanced-panel>
  <gux-tab-advanced-panel tab-id="1-3">
    <!---->
    <div aria-labelledby="gux-1-3-tab" aria-live="assertive" hidden="" id="gux-1-3-panel" role="tabpanel" tabindex="0">
      <span>
        Tab content 3
      </span>
    </div>
  </gux-tab-advanced-panel>
</gux-tabs-advanced>
`;
